metrics/density/docker_memory_usage.sh

```diff
@@ -0,0 +1,318 @@
+#!/bin/bash
+# Copyright (c) 2017-2018 Intel Corporation
+# 
+# SPDX-License-Identifier: Apache-2.0
+#
+#  Description of the test:
+#  This test launches a number of containers in idle mode,
+#  It will then sleep for a configurable period of time to allow
+#  any memory optimisations to 'settle, and then checks the
+#  amount of memory used by all the containers to come up with
+#  an average (using the PSS measurements)
+#  This test uses smem tool to get the memory used.
+
+set -e
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+# Busybox image: Choose a small workload image, this is
+# in order to measure the runtime footprint, not the workload
+# footprint.
+IMAGE='busybox'
+
+CMD='sh'
+NUM_CONTAINERS="$1"
+WAIT_TIME="$2"
+AUTO_MODE="$3"
+TEST_NAME="memory footprint"
+SMEM_BIN="smem"
+KSM_ENABLE_FILE="/sys/kernel/mm/ksm/run"
+
+# Show help about this script
+help(){
+cat << EOF
+Usage: $0 <count> <wait_time> [auto]
+   Description:
+        <count>      : Number of Clear Containers to run.
+        <wait_time>  : Time in seconds to wait before taking
+                       metrics.
+        [auto]       : Optional 'auto KSM settle' mode
+                       waits for ksm pages_shared to settle down
+EOF
+}
+
+# See if KSM is enabled.
+# If so, ammend the test name to reflect that
+check_for_ksm(){
+	if [ ! -f ${KSM_ENABLE_FILE} ]; then
+		return
+	fi
+
+	ksm_on=$(< ${KSM_ENABLE_FILE})
+
+	if [ $ksm_on == "1" ]; then
+		TEST_NAME="${TEST_NAME} ksm"
+	fi
+}
+
+# This function measures the PSS average
+# memory about the child process of each
+# docker-containerd-shim instance in the
+# system.
+get_runc_pss_memory(){
+        avg=0
+        docker_shim="docker-containerd-shim"
+        mem_amount=0
+        count=0
+
+	#FIXME - it would be nice to spit out this data into
```

Comment:
Can you open an issue for this and add it here?

Reply:
Done. #232